src/formatDate.js

```diff
@@ -8,7 +8,68 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  let newFormat = '';
+  const fromSeparator = fromFormat[fromFormat.length - 1];
+  const toSeparator = toFormat[toFormat.length - 1];
+
+  let year = '';
+  let month = '';
+  let day = '';
+
+  let count = 0;
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    switch (fromFormat[i]) {
+      case 'YY':
+        year = date.slice(count, count + fromFormat[i].length);
+
+        if (+year < 30) {
+          year = 20 + year;
+        } else {
+          year = 19 + year;
```

Comment:
The direct concatenation of strings with the `year` variable will not work as expected because `year` is not converted to a string before concatenation, leading to an incorrect year calculation. You should convert the `year` variable to an integer before adding it to `20` or `19`. For example, you could use `year = '19' + year.padStart(2, '0');` for the `19YY` case.